apps/web/src/pages/Profile/EmploymentEquityPage/components/EmploymentEquityForm/EmploymentEquityForm.tsx

```diff
@@ -29,8 +30,13 @@ const EmploymentEquityForm: React.FC<EmploymentEquityFormProps> = ({
 }) => {
   const intl = useIntl();
   const paths = useRoutes();
-  const returnRoute = application
-    ? paths.reviewApplication(application.id)
+  const [searchParams] = useSearchParams();
+  const applicationId = searchParams.get("applicationId");
+  const applicationParam = applicationId
+    ? `?applicationId=${applicationId}`
+    : ``;
+  const returnRoute = applicationId
+    ? paths.reviewApplication(applicationId)
     : paths.profile(user.id);
```

Comment:
I'm tempted to say we make this a custom hook since this seems to be a patter we use in every profile form 🤔 Maybe like:

```tsx
const { param, id } = useCurrentApplication();
```
This would be out of scope for this PR but if you agree we could add a ticket.

Reply:
I thought about the same thing when doing this but wanted to stay within the scope. Feel free to create a new issue with the proposed solution 😄

Reply:
Added the issue, hopefully I summarized it well 😅 

https://github.com/GCTC-NTGC/gc-digital-talent/issues/5609